OpenSim/Region/CoreModules/Scripting/LSLHttp/UrlModule.cs

```diff
@@ -518,11 +518,16 @@ public void AsyncHttpRequest(IHttpServer server, string path, OSHttpRequest http
                     }
                 }
 
+                // Grab the raw unprocessed original query string, if any.
+                int rawQueryPos = httpRequest.Url.Query.IndexOf('?');
+                string rawQueryStr = (rawQueryPos < 0) ? httpRequest.Url.Query : httpRequest.Url.Query.Substring(rawQueryPos + 1);
+
                 //if this machine is behind DNAT/port forwarding, currently this is being
                 //set to address of port forwarding router
                 requestData.headers["x-remote-ip"] = httpRequest.RemoteIPEndPoint.ToString();
                 requestData.headers["x-path-info"] = pathInfo;
-                requestData.headers["x-query-string"] = queryString;
+                requestData.headers["x-query-string"] = rawQueryStr;        // raw original (SL-compatible)
+                requestData.headers["x-query-string-compat"] = queryString; // processed (old Halcyon scripts)
```

Comment:
Shorter answer: I don't have any reason at all to avoid providing a backwards-compatible option. Thinking nobody will use it isn't a good reason not to offer it. It's one well-tested line of code (what we always did). Is there any actual _problem_ with providing it in case it's needed? I don't see one.

For code review, I think the changes to the existing query string option is the significant case. It's an incompatible change.  However, it's currently triggering exceptions and 500 errors and producing the wrong results when compared to SL.  I only know about it because we have a specific case where a properly written script is failing in IW due to the bugs in the former implementation.  So I don't see any option there except to make it work, which also means make it different.  But we can at least provide the old string as well, minus the exceptions and 500s.